presto-main/src/main/java/io/prestosql/sql/gen/BytecodeUtils.java

```diff
@@ -345,6 +350,9 @@ else if (type == ConnectorSession.class) {
         if (functionMetadata.isNullable()) {
             block.append(unboxPrimitiveIfNecessary(scope, returnType));
         }
+        if (Object.class.equals(returnType)) {
```

Comment:
Why equals? Would "object.class.isAssignableFrom returntype && returntype != sqlReturnType.getJavaType()" be better?

Reply:
This cast isn't in the PR any more, but yes, that probably would have been better.